### DISCUSSION TITLE
Auto autocomplete

### DISCUSSION THREAD

**Discussion by lua-rocks** [orbitalquark/textadept#91] [Q&A] [answered]
I like word autocomplete by Ctrl+Enter, but I don't like to press Ctrl+Enter. I want to make this function automatically executable each time when I press any key, but I don't know how to do it.
```lua
  textadept.editing.autocomplete_all_words = true
  events.connect(events.KEYPRESS, function()
    -- Autocomplete current word please!
  end)
```

**Comment by orbitalquark** [comment, marked as answer]
Try this:

    events.connect(events.CHAR_ADDED, function() textadept.editing.autocomplete('word') end)

You will probably want to turn off `buffer.auto_c_choose_single`.

**Reply by lua-rocks** [reply]
Awesome! Thank you very much!
